src/main/scala/nodes/learning/LeastSquaresEstimator.scala

```diff
@@ -0,0 +1,87 @@
+package nodes.learning
+
+import breeze.linalg._
+import nodes.util.{Densify, Sparsify}
+import org.apache.spark.rdd.RDD
+import pipelines.Logging
+import workflow.{Pipeline, WeightedNode, LabelEstimator, OptimizableLabelEstimator}
+
+import scala.reflect._
+
+/**
+ * A least squares solver that is optimized to use a fast algorithm, based on characteristics of the
+ * workload and cost models.
+ *
+ * Currently selects between Dense LBFGS, Sparse LBFGS, Exact Distributed Solve, and Approximate Block Solve.
+ *
+ * The default weights were determined empirically via results run on a 16 r3.4xlarge node cluster.
```

Comment:
Here's a gist with the code I used: https://gist.github.com/etrain/7486bdf1e7b29e381a17

And the data: https://gist.github.com/etrain/f9b05ea7e882f4740063

One important thing about this is instead of doing 
`totalTime = max(k1*cpuTime, k2*memTime) + k3*networkTime`
it estimates
`totalTime = k1*cpuTime + k2*memTime + k3*networkTime` 

But in most cases (e.g. memory and cpu are not well balanced) these things should be functionally equivalent.
Also - I'm a little worried that minimizing least squares loss biases us toward fitting constants in regimes that take a long time, but again, this seems fine and the important bit - that this seems to make good predictions - is maintained.